### PR TITLE
Unit tests fakes

### PR DESCRIPTION
**Pull request by tumf87**
CRUD Panel fake columns and fields traits unit tests.

@tabacitu, this also marks the first PR for tests involving database configuration. 
As this brings in a lot of new and sensible stuff, feedback from you and the community is greatly appreciated.

For more details about how to set up a testing environment involving a database connection, you can check the [PHPUnit guidelines](https://phpunit.de/manual/current/en/database.html).

I've also used orchestral's testbench and database helper libraries. You can check out the documentation and examples for it [here](https://github.com/orchestral/testbench).

### CODE REVIEW

File: tests/Unit/CrudPanel/BaseDBCrudPanelTest.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @return void
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        // call migrations specific to our tests
+        $this->loadMigrationsFrom([
+            '--database' => 'testing',
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/BaseDBCrudPanelTest.php]: That seems to be a reasonable name and I doubt anyone would have a production database called 'testing' (but I've seen stupider things TBH).

File: tests/Unit/CrudPanel/CrudPanelColumnsTest.php [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+        $this->assertEquals(2, count($this->crudPanel->columns));
+        $this->assertEquals($this->expectedTwoColumnsArray, $this->crudPanel->columns);
+    }
+
+    public function testAddColumnNotArray()
+    {
+        $this->setExpectedException(\ErrorException::class);
+
+        $this->crudPanel->addColumns('column1');
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/CrudPanelColumnsTest.php]: Is this missing an assert?

File: tests/Unit/CrudPanel/CrudPanelFakeColumnsTest.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+use Backpack\CRUD\Tests\Unit\Models\Article;
+
+class CrudPanelFakeColumnsTest extends BaseDBCrudPanelTest
+{
+    private $emptyFakeColumnsArray = ['extras'];
+
+    private $fakeFieldsArray = [
+        [
+            'name' => 'field',
+            'label' => 'Normal Field',
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/CrudPanelFakeColumnsTest.php]: I wonder if it is a valid test to see what the world happens if we don't set a label at all seeing as that's actually optional.

File: tests/Unit/Models/Article.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+class Article extends Model
+{
+    use CrudTrait;
+
+    protected $table = 'articles';
+
+    /**
+     * Get the author for the article.
+     */
+    public function user()
```
Comment (lloy0076) [inline tests/Unit/Models/Article.php]: An author called `user` 👿

File: tests/Unit/CrudPanel/BaseDBCrudPanelTest.php (hunk 1 above)
Comment (tumf87) [reply tests/Unit/CrudPanel/BaseDBCrudPanelTest.php]: The database is an in memory one, first created when you run the tests and destroyed immediately after.
I've configured this [here](https://github.com/Laravel-Backpack/CRUD/pull/930/files#diff-1f8df59f532c3641c2e5e010b4d8fa1fR43) using [orchestral's testbench](https://github.com/orchestral/testbench) database helper. So there should be no problem.

Also, I was thinking of suggesting we add an git ignore entry for the tests folder on the release branch. Thoughts?

File: tests/Unit/CrudPanel/CrudPanelColumnsTest.php (hunk 2 above)
Comment (tumf87) [reply tests/Unit/CrudPanel/CrudPanelColumnsTest.php]: Nope, this test method expects an exception. If the exception is not thrown, the test method will fail.

File: tests/Unit/CrudPanel/CrudPanelFakeColumnsTest.php (hunk 3 above)
Comment (tumf87) [reply tests/Unit/CrudPanel/CrudPanelFakeColumnsTest.php]: There are a number of tests doing just that with other input data sets.

File: tests/Unit/CrudPanel/BaseDBCrudPanelTest.php (hunk 1 above)
Comment (lloy0076) [reply tests/Unit/CrudPanel/BaseDBCrudPanelTest.php]: Ah, missed that.

File: tests/Unit/CrudPanel/CrudPanelColumnsTest.php (hunk 2 above)
Comment (lloy0076) [reply tests/Unit/CrudPanel/CrudPanelColumnsTest.php]: Yep, sorry was speed reviewing half asleep.

File: tests/Unit/CrudPanel/CrudPanelFakeColumnsTest.php (hunk 3 above)
Comment (lloy0076) [reply tests/Unit/CrudPanel/CrudPanelFakeColumnsTest.php]: Ok.

**Pull request merged**